loadsbroker/aws.py

```diff
@@ -279,23 +276,23 @@ def update_state(inst):
         while time.time() < end_time and pending:
             self.debug('%d pending instances.' % len(pending))
             # Update the state of all the pending instances
-            yield [self.execute(update_state, inst) for inst in pending]
+            await gen.multi(
+                [self.execute(update_state, inst) for inst in pending])
             pending = self.pending_instances()
 
             # Wait if there's pending to check again
             if pending:
-                yield self.wait(interval)
+                await self.wait(interval)
 
         # Remove everything that isn't running by now
         dead = self.dead_instances() + self.pending_instances()
 
         # Don't wait for the future that kills them
         self.debug("Removing %d dead instances that wouldn't run" % len(dead))
-        self.remove_instances(dead)
+        gen.convert_yielded(self.remove_instances(dead))
```

Comment:
Yea, we spoke about this on IRC: calling a tornado coroutine returns a Future and actually fires it off for one iteration immediately (as an optimization), and it also becomes scheduled.

convert_yielded will convert this native coroutine to a tornado Future (basically wraps it in @gen.coroutine that calls it) so it becomes scheduled like was previously happening

In asyncio you'd instead do a: loop.create_task(coroutine) to schedule it, returning a Future (Task)